src/optimizers/bff_optimizer.py

```diff
@@ -4,7 +4,7 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-# Flexible_Precision_AdamW: a flexible precision AdamW optimizer
+# FlexiblePrecision_AdamW: a flexible precision AdamW optimizer
```

Comment:
probably best not to mix CamelCase with foo_bar styles for class names, no?

Reply:
Hi @stas00  - yes you are right.  I checked the PEP8 guidelines and it is a no go:
"Class | Start each word with a capital letter. **Do not separate words with underscores.**"
I was going for better readability by breaking apart things, but will keep it correct.
Thanks for flagging this!

Reply:
Hi @stas00  - after discussion with @rohan-varma, we are going with "MixedPrecisionAdamW".
This actually meshes well, as the main use case is with FSDP Mixed Precision, and based on some research I've been doing, the easy "I just want to change two lines and free up GPU memory and train faster" is simply plugging it in with the default mixed precision settings. 
You can still set it to pure mode and run all BF16 and with or without Kahan summation for extra precision. 
Thanks again for all the feedback here!

Reply:
This might get confused with the mixed precision concept of AMP, and it's not quite the same conceptually. The concept of mixed precision refers to the different parts of the fwd/bwd/step stages performed in 2 different precisions.

I have absolutely no attachment to the name I proposed earlier, but I'm suggesting to use any new name with self-explainable mnemonics, rather than overload the existing one. 

And since it supports any precision, including FP32, it's hard to see how this is mixed.

The logical concept here is more ANY, rather than AND - perhaps `AnyPrecisionAdamW`

Reply:
Hi @stas00  - agreed, and we're changing to your proposed "AnyPrecisionAdamW" :) 
Thanks again for all the feedback on this!